code_generation/templates/power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_gen/attribute_classes.hpp.jinja

```diff
@@ -29,14 +30,12 @@ struct get_attributes_list<{{ attribute_class.name }}<sym>> {
 template<>
 struct get_attributes_list<{{ attribute_class.name }}> {
 {%- endif %}
-    std::vector<MetaAttribute> operator() () const {
-        // all attributes including base class
-        return {
+    static constexpr std::array<MetaAttribute, {{ attribute_class.full_attributes|length }}> value{
+            // all attributes including base class
             {% for attribute in attribute_class.full_attributes %}
-            {MetaAttributeImpl<{{ attribute_class.full_name }}, &{{ attribute_class.full_name }}::{{ attribute.names }}>{}, "{{ attribute.names }}"},
+            meta_data_gen::get_meta_attribute<{{ attribute_class.full_name }}, &{{ attribute_class.full_name }}::{{ attribute.names }}, offsetof({{ attribute_class.full_name }}, {{ attribute.names }}), []{ return "{{ attribute.names }}"; }>::value,
```

Comment:
this line becomes incredibly long for some components. can you maybe for readability put the attributes on new lines?

Reply:
Again. This is generated code. Most of the time, no one (including PGM developers) will read it.